Second phase of hipTensor docs refresh (#384)
ROCm/hipTensor: docs/contribution/contributors-guide.rst

Our code contribution guidelines closely follows the model of `GitHub
pull-requests <https://help.github.com/articles/using-pull-requests/>`__.

> The hipTensor repository uses a workflow with a ``master`` branch for releases and a

Reviewer: The release branch isn't master anymore. It's in the format release\rocm-rel-x.y. You can probably just call it a ``release`` branch.

``develop`` branch for integrating new code. Follow the guidelines below when creating a pull request:

-  Target the ``develop`` branch for integration.
-  Ensure that the code builds successfully.
-  Do not break existing test cases.
-  Be informed that a new functionality is only merged with new unit tests.
-  Ensure that new unit tests integrate within the existing GoogleTest framework.
-  Design the tests with thorough code coverage.
-  Include benchmark tests and ensure performance approaches the compute-bound or memory-bound limits.

Style guide
============

This project follows the `CPP Core
guidelines <https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md>`__,

> with few modifications and additions listed below. We encourage you to follow these guidelines when creating pull requests.

Reviewer: ```suggestion
with a few modifications and additions listed below. Follow these guidelines when creating pull requests.
```

Implementation
--------------

-  Use the ``.cpp`` suffix for code files and ``.hpp`` suffix for interface files, if your project doesn't already follow another
   convention.
-  Ensure that each ``.cpp`` file includes the ``.hpp`` file(s) that defines its interface.
-  Avoid placing global ``using`` directives in header files.
-  Use ``#include`` guards for all ``.hpp`` files.
-  Avoid using unnamed (anonymous) ``namespace``s in header files.
-  Use ``std::array`` or ``std::vector`` instead of C arrays.
-  Minimize the exposure of class members.
-  Keep functions short and simple.
-  To return multiple output values, prefer using a ``std::tuple``.
-  Manage resources automatically using RAII, including ``std::unique_ptr`` and ``std::shared_ptr``.
-  Use ``auto`` to avoid redundant repetition of type names.
-  Always initialize an object.
-  Use the ``{}`` initializer syntax wherever possible.
-  Expect your code to run as part of a multi-threaded program.
-  Avoid using global variables.

Format
------

C++ code is formatted using ``clang-format``. To run ``clang-format``,
use the version in the ``/opt/rocm/llvm/bin`` directory. Avoid using your

> system's built-in ``clang-format``, as it may be outdated and produce inconsistent results.

Reviewer: We typically use 'might' here instead of 'may', which implies permission.

1. Create and track a hipTensor fork.
2. Clone your fork:

   .. code-block:: bash

      git clone -b develop https://github.com/<your_fork>/hipTensor.git .
      .githooks/install
      git checkout -b <new_branch>
      ...
      git add <new_work>
      git commit -m "What was changed"
      git push origin <new_branch>
      ...

> 3. Create a pull request to ``ROCmSoftwarePlatform/hipTensor`` targeting the ``develop`` branch.

Reviewer: ```suggestion
3. Create a pull request to ``ROCm/hipTensor``, targeting the ``develop`` branch.
```
